src/librustc/traits/object_safety.rs

```diff
@@ -354,7 +354,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                     // direct equality here because all of these types
                     // are part of the formal parameter listing, and
                     // hence there should be no inference variables.
-                    let projection_trait_ref = ty::Binder(data.trait_ref.clone());
+                    let projection_trait_ref = ty::Binder(data.trait_ref(self).clone());
```

Comment:
Cloning is silly on `Copy` types - this stuff predates `Copy` `Substs`, remove on sight.